examples_test.go

```diff
@@ -0,0 +1,65 @@
+package spdx_go_test
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/spdx/spdx-go-model/spdx_v3_0_1"
+)
+
+func TestImport(t *testing.T) {
+	spdx_v3_0_1.MakePerson()
+}
+
+func TestExample(t *testing.T) {
+	// Create a new Person
+	p := spdx_v3_0_1.MakePerson()
+
+	// Set the unique identifier (IRI) for the person
+	p.ID().Set("http://spdx.org/spdxdocs/person-example")
+
+	// Set the name for the person
+	p.Name().Set("Joshua")
+
+	// Create a creation info object
+	ci := spdx_v3_0_1.MakeCreationInfo()
+	ci.SpecVersion().Set("3.0.1")
+	ci.Created().Set(time.Now())
```

Comment:
Avoid Unnecessary [Getters](https://go.dev/doc/effective_go#Getters) 
 - Instead of using `p.ID().Set(value)`, use `p.SetID(value)`,
 - Similarly, `p.Name().Set(value)`, use `p.SetName(value)`
 - Similarly, `ci.SpecVersion().Set(value)`, use `ci.SetSpecVersion(value)`
 - and, `ci.Created().Set(valu)`, use `ci.SetCreated(value)`
 
 Something like this: https://go.dev/play/p/tREj7_5eQFN
 
You can checkout these resources to understand more about getter and setter in go:
- https://stackoverflow.com/questions/11810218/how-to-set-and-get-fields-in-structs-method/11842484#11842484
- https://softwareengineering.stackexchange.com/questions/278807/does-go-implicitly-discourage-getters-and-setters-and-encourage-direct-member-ac

Reply:
The reason for this is because all of this code is auto-generated, so having a common "string property" class that can be reused all the places where a class has a string property is much easier to generate code for than (effectively) open coding it everywhere. I do appreciate this is not "the go way", so I'll look into it though